### PR TITLE
Allow custom name mapping for discovered models

### PR DESCRIPTION
**Pull request by raymondfeng**

### CODE REVIEW

**Comment by bajtos** [comment]
Please add unit-tests to cover the two new options:
 - `options.nameMapper === null`
 - `typeof options.nameMapper === 'function'`

**Comment by raymondfeng** [comment]
@bajtos Tests added

File: test/discover.test.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+        done();
+      });
+    });
+
+});
+
+
+
+
+
```
Comment (bajtos) [inline test/discover.test.js]: Please remove these extra empty lines.

**Comment by bajtos** [comment]
Perhaps `test/discovery.test.js` would be a better filename than `test/discover.test.js`?

**Comment by bajtos** [comment]
Two minor comments, the rest LGTM. You can land the patch after you address my first comment. Feel free to ignore the second one.

**Pull request merged**